daemon/libnetwork/internal/resolvconf/resolvconf.go

```diff
@@ -419,10 +455,13 @@ func UserModified(rcPath, rcHashPath string) (bool, error) {
 }
 
 func (rc *ResolvConf) processLine(line string) {
-	fields := strings.Fields(line)
-
 	// Strip blank lines and comments.
-	if len(fields) == 0 || fields[0][0] == '#' || fields[0][0] == ';' {
+	if line == "" || line[0] == '#' || line[0] == ';' {
+		return
+	}
+
+	fields := strings.Fields(line)
+	if len(fields) == 0 {
```

Comment:
Pushed an extra commit for these; it occurred to me that we were calling `string.Fields()` before checking if it was an empty line or comment; better to skip early.